### ISSUE TITLE
Reranker 2: Train neural reranker with frozen embeddings

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#630]
# Task

Train neural reranker for query-document relevance scoring.

## Parent Epic

Part of **#616** (EPIC: Tier-Filtered Model Retraining)

## Model Architecture

~180K parameter model:
```python
class Reranker(nn.Module):
    def __init__(self, compositional_embedding):
        self.comp_emb = compositional_embedding  # Frozen
        self.scorer = nn.Sequential(...)         # Learned scoring
```

## Training Configuration

- **Input**: Query + document embeddings (frozen compositional)
- **Output**: Relevance score [0, 1]
- **Loss**: Binary cross-entropy
- **Epochs**: 20 (early stopping patience=5)
- **Batch size**: 64
- **Learning rate**: 0.0001

## Training Data

From **#629**: `data/training/reranker_pairs.jsonl`

## Output

File: `models/reranker/best_model.pt`

## Acceptance Criteria

- [ ] Model uses frozen root embeddings
- [ ] Training converges
- [ ] Validation accuracy > 80%
- [ ] Model saved with checkpoint

## Command

```bash
./scripts/train_reranker.sh --fresh
```

## Dependencies

- Blocked by: #629 (Generate training data)
- Blocks: #631 (Test reranker)

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
